
# HG changeset patch
# User Daniel Burrows <dburrows@debian.org>
# Date 1175315176 25200
# Node ID d5ab883ef90a0e4ea13985d9db90ddbb84441ad5
# Parent dc54ae48d8c120cf268eea375e373471d502722a
Fix handling of DB4 hcache open failure. (closes: #2714)

--- a/hcache.c	Fri Mar 30 20:18:02 2007 -0700
+++ b/hcache.c	Fri Mar 30 21:26:16 2007 -0700
@@ -984,17 +984,13 @@ mutt_hcache_open(const char *path, const
 
   ret = (*h->env->open)(h->env, NULL, DB_INIT_MPOOL | DB_CREATE | DB_PRIVATE,
 	0600);
-  if (!ret)
-  {
-    ret = db_create(&h->db, h->env, 0);
-    if (ret)
-    {
-      h->env->close(h->env, 0);
-      mx_unlock_file(h->lockfile, h->fd, 0);
-      close(h->fd);
-      FREE(&h);
-      return NULL;
-    }
+  if (ret)
+  {
+    h->env->close(h->env, 0);
+    mx_unlock_file(h->lockfile, h->fd, 0);
+    close(h->fd);
+    FREE(&h);
+    return NULL;
   }
 
   if (stat(path, &sb) != 0 && errno == ENOENT)

